### ISSUE TITLE
Do not scroll List items unless selected item is out of view

### ISSUE THREAD

**Issue by ardnew** [rivo/tview#217]
When navigating a `List` with more items than can fit on screen, the `Draw` routine tries to simulate a scrolling effect. However, this scrolling only works as expected in one direction -- when you scroll down.

When the user scrolls the list downward, the bottom-most item is highlighted/selected as the prior items scroll upward out of view _as expected_. 

When the user scrolls the list upward, the bottom-most item is _still_ highlighted/selected as the prior items scroll downward out of view, which is not what I think most people would expect.

For comparison, consider any given text editor. If your cursor is on the bottom-most row of the editor view, and you press the `Up` key on your keyboard, the cursor moves upward and the items visible on screen remain stationary until the cursor reaches the top of the screen -- at which point the whole relationship should happen in reverse.

Basically, unless the user's currently selected item is outside the range of visible items, the view should not be scrolling at all.

**Comment by rivo** [comment]
See also https://github.com/rivo/tview/pull/218#issuecomment-456959828.
